test/modules/candy-machine/initCandyMachine.test.ts

```diff
@@ -0,0 +1,371 @@
+  const [gateKeeper] = amman.genKeypair();
+
+  const { cm, minimalConfig, opts } = await init();
+
+  const config: CandyMachineConfigWithoutStorage = {
+    ...minimalConfig,
+    gatekeeper: { expireOnUse: true, gatekeeperNetwork: gateKeeper.toBase58() },
+  };
+
+  const { transactionId, confirmResponse, candyMachine, ...rest } =
+    await cm.initCandyMachineFromConfig(config, opts);
+  await amman.addr.addLabel('initCandyMachine', transactionId);
+
+  assertConfirmedWithoutError(t, cusper, confirmResponse);
+  assertProperlyInitialized(t, {
+    ...rest,
+    ...config,
+    candyMachine,
+    tokenMint: null,
+  });
+  spok(t, candyMachine.data.gatekeeper as GatekeeperConfig, {
+    $topic: 'gatekeeper',
+    expireOnUse: config.gatekeeper?.expireOnUse,
+    gatekeeperNetwork: spokSamePubkey(config.gatekeeper?.gatekeeperNetwork),
+  });
+});
+
+test('candyMachine: with invalid gatekeeper settings (network not a public key)', async (t) => {
+  const { cm, minimalConfig, opts } = await init();
+
+  const config: CandyMachineConfigWithoutStorage = {
+    ...minimalConfig,
+    gatekeeper: { expireOnUse: true, gatekeeperNetwork: '<invalid>' },
+  };
+
+  await assertThrows(
+    t,
+    () => cm.initCandyMachineFromConfig(config, opts),
+    /not a valid PublicKey/i
+  );
+});
+
+// -----------------
+// WhitelistMint Settings
+// -----------------
+test('candyMachine: with whitelistMint settings', async (t) => {
+  const [mint] = amman.genKeypair();
+
+  const { cm, minimalConfig, opts } = await init();
+
+  const config: CandyMachineConfigWithoutStorage = {
+    ...minimalConfig,
+    whitelistMintSettings: {
+      mode: 'burnEveryTime',
+      discountPrice: 5,
+      mint: mint.toBase58(),
+      presale: false,
+    },
+  };
+
+  const { transactionId, confirmResponse, candyMachine, ...rest } =
+    await cm.initCandyMachineFromConfig(config, opts);
+  await amman.addr.addLabel('initCandyMachine', transactionId);
+
+  assertConfirmedWithoutError(t, cusper, confirmResponse);
+  assertProperlyInitialized(t, {
+    ...rest,
+    ...config,
+    candyMachine,
+    tokenMint: null,
+  });
+  spok(t, candyMachine.data.whitelistMintSettings as WhitelistMintSettings, {
+    $topic: 'whitelist mint settings',
+    mode: WhitelistMintMode.BurnEveryTime,
+    discountPrice: spokSameBignum(config.whitelistMintSettings?.discountPrice!),
+    mint: spokSamePubkey(config.whitelistMintSettings?.mint!),
+    presale: config.whitelistMintSettings?.presale,
+  });
+});
+
+test('candyMachine: with invalid whitemint settings (mint not a public key)', async (t) => {
+  const { cm, minimalConfig, opts } = await init();
+
+  const config: CandyMachineConfigWithoutStorage = {
+    ...minimalConfig,
+    whitelistMintSettings: {
+      mode: 'burnEveryTime',
+      discountPrice: 5,
+      mint: '<invalid mint key>',
+      presale: false,
+    },
+  };
+
+  await assertThrows(
+    t,
+    () => cm.initCandyMachineFromConfig(config, opts),
```

Comment:
There are no tests using `initCandyMachine` directly. I know that `initCandyMachineFromConfig` uses it under the hood but we should at least make sure it works on its own since it's part of the public API.

Reply:
Not a huge fan of duplicating tests.
But I can add just one using it directly just to even ensure its exported I suppose.

Reply:
IMO, `initCandyMachine` should be the one that's heavily tested and `initCandyMachineFromConfig` should have just enough tests to ensure the parsing is correct. That way there's no test duplication, you're testing the specifics of each method.